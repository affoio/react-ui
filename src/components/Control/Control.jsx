import React from 'react';
import styled from 'styled-components';

import { ControlLabel, HelperText } from './ControlText';
import ControlWrapper from './ControlWrapper';
import ControlContainer from './ControlContainer';

const ControlContent = styled.div`
  flex-grow: 1;
  display: flex;
  height: 100%;
  align-items: center;
`;

const Control = props => {
  const { disabled, label, error, focused, after, before, helperText, fullWidth, children, onWrapperClick } = props;
  const { wrapperStyles, onFocus, onBlur, tabIndex } = props;

  return (
    <ControlContainer {...{ fullWidth, disabled }} onBlur={onBlur} onFocus={onFocus} tabIndex={tabIndex}>
      {label && (
        <ControlLabel focused={focused} error={!!error}>
          {label}
        </ControlLabel>
      )}
      <ControlWrapper wrapperStyles={wrapperStyles} error={error} focused={focused} onClick={onWrapperClick}>
        {before}
        <ControlContent>{children}</ControlContent>
        {after}
      </ControlWrapper>

      <HelperText error={error}>
        {!!error && error}
        {helperText && !error && helperText}
      </HelperText>
    </ControlContainer>
  );
};

export default Control;
