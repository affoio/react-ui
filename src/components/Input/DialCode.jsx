import styled from 'styled-components';
import colors from '../../colors';
import { getTypographyFontSizeStyles, getMultipliedIndent } from '../../theme/selectors';

const DialCode = styled.span`
  ${({ theme }) => getTypographyFontSizeStyles(theme, 100)};
  color: ${colors.black[500]};
  margin: 0 ${({ theme }) => getMultipliedIndent(theme, 2)};
`;

export default DialCode;
