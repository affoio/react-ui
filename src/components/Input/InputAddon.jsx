import styled from 'styled-components';
import ComponentAddon from '../ComponentAddon/ComponentAddon';

const InputAddon = styled(ComponentAddon).attrs({
  marginSize: 2,
})`
  width: 24px;
  height: 24px;
`;

export default InputAddon;
