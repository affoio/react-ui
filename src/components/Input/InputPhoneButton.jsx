import styled from 'styled-components';
import Button from '../Button/Button';
import colors from '../../colors';
import { getMultipliedIndent } from '../../theme/selectors';

const InputPhoneButton = styled(Button).attrs({
  use: 'tertiary',
})`
  padding: ${({ theme }) => `0 ${getMultipliedIndent(theme, 4)}`};
  color: ${colors.black[800]};
  height: 56px;
  line-height: 56px;
  margin-top: -1px;
  border-color: transparent;
  border-right: 1px solid ${colors.black[300]};
  margin-right: ${({ theme }) => `${getMultipliedIndent(theme, 4)}`};

  &:hover {
    color: indianred;
    background-color: transparent;
  }

  &:focus {
    color: indianred;
    background-color: transparent;
  }
`;

export default InputPhoneButton;
