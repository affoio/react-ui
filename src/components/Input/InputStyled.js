import styled from 'styled-components';
import colors from '../../colors';
import { addStyleIfPropTruly } from '../../helpers/styledComponents';

export const InputStyled = styled.input`
  outline: none;
  border: none;
  background-color: transparent;
  flex-grow: 1;
  height: 100%;
  box-sizing: border-box;
  font-size: inherit;
  max-width: 100%;
  width: 100%;

  &::placeholder {
    color: ${colors.black[400]};
  }

  &::-ms-clear,
  &::-ms-reveal {
    display: none;
  }

  ${addStyleIfPropTruly('readOnly', 'cursor: pointer;')}
`;
