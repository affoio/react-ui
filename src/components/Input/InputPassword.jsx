import React, { useState } from 'react';
import styled from 'styled-components';
import Input from './Input';
import CloseEyeIcon from '../Icon/CloseEyeIcon';
import OpenEyeIcon from '../Icon/OpenEyeIcon';
import colors from '../../colors';

const SwitchEyeStyled = styled.div`
  cursor: pointer;
`;

export const SwitchEye = ({ onClick, type }) => {
  return (
    <SwitchEyeStyled onClick={onClick}>
      {type === 'password' ? (
        <CloseEyeIcon color={colors.black[400]} hoverColor={colors.black[600]} />
      ) : (
        <OpenEyeIcon color={colors.black[400]} hoverColor={colors.black[600]} />
      )}
    </SwitchEyeStyled>
  );
};

const InputPassword = props => {
  const [type, changeType] = useState('password');
  const AfterComponent = (
    <SwitchEye type={type} onClick={() => changeType(prevType => (prevType === 'password' ? 'text' : 'password'))} />
  );

  return <Input {...props} type={type} after={AfterComponent} />;
};

export default InputPassword;
