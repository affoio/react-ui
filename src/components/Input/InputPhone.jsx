import React, { useState, useEffect, useCallback, useRef, useMemo } from 'react';
import { css } from 'styled-components';
import { pipe } from 'ramda';
import Input from './Input';
import Select from '../Select/Select';
import InputPhoneButton from './InputPhoneButton';
import countries from './countries/countries';
import Flag from '../Flag/Flag';
import SelectItem from '../Select/SelectItem';
import DialCode from './DialCode';
import ComponentAddon from '../ComponentAddon/ComponentAddon';

const guessSelectedCountry = (inputNumber, onlyCountries, defaultCountry) => {
  if (inputNumber.trim() === '') return defaultCountry;

  const bestGuess = onlyCountries.reduce(
    (selectedCountry, country) => {
      if (inputNumber.startsWith(country.value)) {
        if (country.value.length > selectedCountry.value.length) {
          return country;
        }
        if (country.value.length === selectedCountry.value.length && country.priority < selectedCountry.priority) {
          return country;
        }
      }
      return selectedCountry;
    },
    { value: '', priority: 10001 }
  );

  if (!bestGuess.value) return defaultCountry || { value: '' };

  return bestGuess;
};

export const defaultDialCode = '+44';
const defaultSelectValue = defaultNumber => {
  const findOption = guessSelectedCountry(defaultNumber, countries);
  const findOptionIndex = countries.findIndex(
    option => findOption.value === option.value && option.priority === findOption.priority
  );

  return {
    option: findOption,
    index: findOptionIndex,
  };
};

const wrapperStyles = css`
  padding-left: 0;
`;

const addLeadingPlusIfEmpty = num => (num.length ? num : '+');
const updateValueOnlyIfMatch = value => num => (num.match(/^\+[0-9]*$/) ? num : value);

const normalizePhoneValue = (value, prevValue) => {
  const updatePrevValue = updateValueOnlyIfMatch(prevValue);

  return pipe(
    addLeadingPlusIfEmpty,
    updatePrevValue
  )(value);
};

const ToggleComponent = ({ selectedOption, arrow, ...other }) => (
  <InputPhoneButton after={arrow} {...other}>
    <Flag country={selectedOption.iso2} />
  </InputPhoneButton>
);

const OptionComponent = ({ option, ...otherOptionProps }) => (
  <SelectItem {...otherOptionProps}>
    <ComponentAddon marginSize={4}>
      <Flag country={option.iso2} />
    </ComponentAddon>
    <div>
      {option.label}
      <DialCode>{option.value}</DialCode>
    </div>
  </SelectItem>
);

const BeforeComponent = React.memo(
  ({ onChange, selectedValue = {}, defaultValue, focusedValue, onChangeFocusedOption }) => (
    <Select
      toggle={ToggleComponent}
      option={OptionComponent}
      onChange={onChange}
      options={countries}
      selectedValue={selectedValue}
      defaultValue={defaultValue}
      focusedValue={focusedValue}
      onChangeFocusedOption={index => onChangeFocusedOption(index)}
      inline
    />
  )
);

const InputPhone = props => {
  const { country, onChangeCountry, onChange, defaultValue, ...otherProps } = props;
  const finalDefaultValue = defaultValue || defaultDialCode;
  const defaultOptionValue = useMemo(() => defaultSelectValue(finalDefaultValue), [finalDefaultValue]);
  const [value, changeValue] = useState(finalDefaultValue);
  const [selectedValue, onChangeSelectedValue] = useState(defaultOptionValue.option);
  const [focusedValue, onChangeFocusedValue] = useState(defaultOptionValue.index);
  const optionFromCountry = country ? countries.find(option => option.iso2 === country) : null;
  const optionFromCountryIndex = country ? countries.findIndex(option => option.iso2 === country) : null;

  const mainSelectedValue = optionFromCountry || selectedValue;
  const onChangeHandler = useCallback(
    e => {
      const number = normalizePhoneValue(e.target.value, value);
      if (number !== value) {
        changeValue(number);

        const findOption = guessSelectedCountry(number, countries, mainSelectedValue);
        const findOptionIndex = countries.findIndex(
          option => findOption.value === option.value && option.priority === findOption.priority
        );
        if (onChangeCountry) {
          onChangeCountry(findOption);
        }
        onChangeSelectedValue(findOption || mainSelectedValue);
        if (findOptionIndex > -1) {
          onChangeFocusedValue(findOptionIndex);
        }
        if (onChange) {
          onChange(number);
        }
      }
    },
    [mainSelectedValue, onChange, onChangeCountry, value]
  );

  const inputRef = useRef();

  const onSelectHandler = useCallback(
    option => {
      changeValue(option.value);
      onChangeSelectedValue(option);
      if (onChangeCountry) {
        onChangeCountry(option);
      }
      inputRef.current.focus();
    },
    [onChangeCountry]
  );

  useEffect(() => {
    if (optionFromCountry) {
      changeValue(optionFromCountry.value);
      onChangeSelectedValue(optionFromCountry);
      onChangeFocusedValue(optionFromCountryIndex);
    }
  }, [country, optionFromCountry, optionFromCountryIndex]);

  return (
    <Input
      {...otherProps}
      onChange={onChangeHandler}
      wrapperStyles={wrapperStyles}
      maxLength={20}
      beforeAddon={
        <BeforeComponent
          selectedValue={mainSelectedValue}
          focusedValue={focusedValue}
          defaultValue={defaultOptionValue.option}
          onChange={onSelectHandler}
          onChangeFocusedOption={onChangeFocusedValue}
        />
      }
      value={value}
      ref={inputRef}
    />
  );
};

export default InputPhone;
