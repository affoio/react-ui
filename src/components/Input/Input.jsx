import React, { useState } from 'react';
import { string, func, bool, element } from 'prop-types';
import { omit } from 'ramda';
import { InputStyled } from './InputStyled';
import Control from '../Control/Control';
import InputAddon from './InputAddon';

export const defaultProps = {
  type: 'text',
  defaultValue: '',
};

export const propTypes = {
  error: string,
  label: string,
  helperText: string,
  onFocus: func,
  onBlur: func,
  disabled: bool,
  onChange: func,
  defaultValue: string,
  before: element,
  after: element,
  type: string,
  readOnly: bool,
};

const Input = React.forwardRef((props, ref) => {
  const { value, defaultValue, error, label, helperText, disabled } = props;
  const { onChange, onFocus, onBlur, before, after, readOnly, beforeAddon } = props;
  const { wrapperStyles } = props;
  const [stateValue, changeValue] = useState(defaultValue);
  const [focused, changeFocus] = useState(false);
  const inputValue = typeof value !== 'undefined' ? value : stateValue;
  const onFocusHandler = e => {
    if (onFocus) {
      onFocus(e);
    }

    changeFocus(true);
  };

  const onBlurHandler = e => {
    if (onBlur) {
      onBlur(e);
    }

    changeFocus(false);
  };

  const onChangeHandler = e => {
    if (onChange) {
      onChange(e);
    }

    changeValue(e.target.value);
  };

  return (
    <Control
      {...{ disabled, label, error, focused, helperText }}
      fullWidth
      before={beforeAddon || (before && <InputAddon>{before}</InputAddon>)}
      after={after && <InputAddon position="right">{after}</InputAddon>}
      wrapperStyles={wrapperStyles}
    >
      <InputStyled
        {...omit(['defaultValue'], props)}
        onFocus={onFocusHandler}
        onBlur={onBlurHandler}
        onChange={onChangeHandler}
        value={inputValue}
        readOnly={readOnly}
        ref={ref}
      />
    </Control>
  );
});

Input.propTypes = propTypes;
Input.defaultProps = defaultProps;

export default React.memo(Input);
