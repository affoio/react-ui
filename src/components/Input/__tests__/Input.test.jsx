import React from 'react';
import { render, mount } from 'enzyme';
import Input from '../Input';
import SearchIcon from '../../Icon/SearchIcon';
import { InputStyled } from '../InputStyled';
import Control from '../../Control/Control';

describe('<Input>', () => {
  describe('Input style snapshots', () => {
    it('renders correctly', () => {
      const tree = render(<Input />);

      expect(tree).toMatchSnapshot();
    });
    it('renders correctly with label', () => {
      const wrapper = mount(<Input label="Some label" />);

      const controlWrapper = wrapper.find(Control);

      expect(controlWrapper.length).toBe(1);
      expect(controlWrapper.props().label).toBe('Some label');
    });

    it('renders correctly with helperText', () => {
      const wrapper = mount(<Input helperText="Some helper text" />);
      const controlWrapper = wrapper.find(Control);

      expect(controlWrapper.length).toBe(1);
      expect(controlWrapper.props().helperText).toBe('Some helper text');
    });

    it('renders correctly when disabled', () => {
      const tree = render(<Input disabled />);

      expect(tree).toMatchSnapshot();
    });

    it('renders correctly with error', () => {
      const wrapper = mount(<Input error="Oh no error" />);

      const controlWrapper = wrapper.find(Control);
      expect(controlWrapper.length).toBe(1);
      expect(controlWrapper.props().error).toBe('Oh no error');
    });

    it('renders correctly focused label with error', () => {
      const wrapper = mount(<Input error="Oh no error" label="some" />);
      wrapper.find('input').simulate('focus');
      const controlProps = wrapper.find(Control).props();

      expect(controlProps.focused).toBe(true);
      expect(controlProps.error).toBe('Oh no error');
    });

    it('renders correctly focused label', () => {
      const wrapper = mount(<Input label="some" />);
      wrapper.find('input').simulate('focus');

      expect(wrapper.find(Control).props().focused).toBe(true);
    });

    it('renders correctly with beforeElement', () => {
      const tree = render(<Input before={<SearchIcon />} />);

      expect(tree).toMatchSnapshot();
    });

    it('renders correctly with after', () => {
      const tree = render(<Input after={<SearchIcon />} />);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Input handlers', () => {
    it('calls onFocus when it passed', () => {
      const mockFn = jest.fn();
      const wrapper = mount(<Input onFocus={mockFn} />);
      wrapper.find(InputStyled).simulate('focus');

      expect(wrapper.find(Control).props().focused).toBe(true);
      expect(mockFn).toBeCalled();
    });

    it('doesnt calls onFocus when it does not passed', () => {
      const wrapper = mount(<Input />);
      wrapper.find(InputStyled).simulate('focus');

      expect(wrapper.find(Control).props().focused).toBe(true);
    });

    it('calls onBlur when it passed', () => {
      const mockFn = jest.fn();
      const wrapper = mount(<Input onBlur={mockFn} />);
      wrapper.find(InputStyled).simulate('blur');

      expect(wrapper.find(Control).props().focused).toBe(false);
      expect(mockFn).toBeCalled();
    });

    it('doesnt calls onBlur when it does not passed', () => {
      const mockFn = jest.fn();
      const wrapper = mount(<Input />);
      wrapper.find(InputStyled).simulate('blur');

      expect(wrapper.find(Control).props().focused).toBe(false);
      expect(mockFn).not.toBeCalled();
    });

    it('calls onChange when it passed', () => {
      const mockFn = jest.fn();
      const wrapper = mount(<Input onChange={mockFn} />);
      wrapper.find('input').simulate('change', { target: { value: 'Hello' } });

      expect(wrapper.find('input').instance().value).toBe('Hello');
      expect(mockFn).toBeCalled();
    });

    it('doesnt calls onBlur when it does not passed', () => {
      const mockFn = jest.fn();
      const wrapper = mount(<Input />);
      wrapper.find('input').simulate('change', { target: { value: 'Hello' } });

      expect(wrapper.find('input').instance().value).toBe('Hello');
      expect(mockFn).not.toBeCalled();
    });
  });
});
