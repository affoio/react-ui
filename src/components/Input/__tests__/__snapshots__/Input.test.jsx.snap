// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> Input style snapshots renders correctly 1`] = `
.c3 {
  outline: none;
  border: none;
  background-color: transparent;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 100%;
  box-sizing: border-box;
  font-size: inherit;
  max-width: 100%;
  width: 100%;
}

.c3::-webkit-input-placeholder {
  color: #bfbfbf;
}

.c3::-moz-placeholder {
  color: #bfbfbf;
}

.c3:-ms-input-placeholder {
  color: #bfbfbf;
}

.c3::placeholder {
  color: #bfbfbf;
}

.c3::-ms-clear,
.c3::-ms-reveal {
  display: none;
}

.c4 {
  font-size: 12px;
  box-sizing: border-box;
  line-height: 10px;
  padding-left: 16px;
  padding-top: 6px;
  color: #8c8c8c;
  height: 16px;
}

.c1 {
  width: 100%;
  height: 56px;
  box-sizing: border-box;
  background-color: #f2f2f2;
  font-size: 16px;
  color: #000000;
  padding: 0 16px;
  border: 1px solid #f2f2f2;
  border-radius: 2px;
  font-weight: 400;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: block;
  width: 100%;
  position: relative;
  outline: none;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <input
        class="c3"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="c4"
  />
</div>
`;

exports[`<Input> Input style snapshots renders correctly when disabled 1`] = `
.c3 {
  outline: none;
  border: none;
  background-color: transparent;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 100%;
  box-sizing: border-box;
  font-size: inherit;
  max-width: 100%;
  width: 100%;
}

.c3::-webkit-input-placeholder {
  color: #bfbfbf;
}

.c3::-moz-placeholder {
  color: #bfbfbf;
}

.c3:-ms-input-placeholder {
  color: #bfbfbf;
}

.c3::placeholder {
  color: #bfbfbf;
}

.c3::-ms-clear,
.c3::-ms-reveal {
  display: none;
}

.c4 {
  font-size: 12px;
  box-sizing: border-box;
  line-height: 10px;
  padding-left: 16px;
  padding-top: 6px;
  color: #8c8c8c;
  height: 16px;
}

.c1 {
  width: 100%;
  height: 56px;
  box-sizing: border-box;
  background-color: #f2f2f2;
  font-size: 16px;
  color: #000000;
  padding: 0 16px;
  border: 1px solid #f2f2f2;
  border-radius: 2px;
  font-weight: 400;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: block;
  width: 100%;
  opacity: 0.5;
  position: relative;
  outline: none;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
  disabled=""
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <input
        class="c3"
        disabled=""
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="c4"
  />
</div>
`;

exports[`<Input> Input style snapshots renders correctly with after 1`] = `
.c3 {
  outline: none;
  border: none;
  background-color: transparent;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 100%;
  box-sizing: border-box;
  font-size: inherit;
  max-width: 100%;
  width: 100%;
}

.c3::-webkit-input-placeholder {
  color: #bfbfbf;
}

.c3::-moz-placeholder {
  color: #bfbfbf;
}

.c3:-ms-input-placeholder {
  color: #bfbfbf;
}

.c3::placeholder {
  color: #bfbfbf;
}

.c3::-ms-clear,
.c3::-ms-reveal {
  display: none;
}

.c6 {
  font-size: 12px;
  box-sizing: border-box;
  line-height: 10px;
  padding-left: 16px;
  padding-top: 6px;
  color: #8c8c8c;
  height: 16px;
}

.c1 {
  width: 100%;
  height: 56px;
  box-sizing: border-box;
  background-color: #f2f2f2;
  font-size: 16px;
  color: #000000;
  padding: 0 16px;
  border: 1px solid #f2f2f2;
  border-radius: 2px;
  font-weight: 400;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: block;
  width: 100%;
  position: relative;
  outline: none;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 8px;
  width: 24px;
  height: 24px;
}

.c5 {
  fill: currentColor;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <input
        class="c3"
        type="text"
        value=""
      />
    </div>
    <div
      class="c4"
    >
      <svg
        class="c5"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
  </div>
  <div
    class="c6"
  />
</div>
`;

exports[`<Input> Input style snapshots renders correctly with beforeElement 1`] = `
.c5 {
  outline: none;
  border: none;
  background-color: transparent;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 100%;
  box-sizing: border-box;
  font-size: inherit;
  max-width: 100%;
  width: 100%;
}

.c5::-webkit-input-placeholder {
  color: #bfbfbf;
}

.c5::-moz-placeholder {
  color: #bfbfbf;
}

.c5:-ms-input-placeholder {
  color: #bfbfbf;
}

.c5::placeholder {
  color: #bfbfbf;
}

.c5::-ms-clear,
.c5::-ms-reveal {
  display: none;
}

.c6 {
  font-size: 12px;
  box-sizing: border-box;
  line-height: 10px;
  padding-left: 16px;
  padding-top: 6px;
  color: #8c8c8c;
  height: 16px;
}

.c1 {
  width: 100%;
  height: 56px;
  box-sizing: border-box;
  background-color: #f2f2f2;
  font-size: 16px;
  color: #000000;
  padding: 0 16px;
  border: 1px solid #f2f2f2;
  border-radius: 2px;
  font-weight: 400;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: block;
  width: 100%;
  position: relative;
  outline: none;
}

.c4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 8px;
  width: 24px;
  height: 24px;
}

.c3 {
  fill: currentColor;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <svg
        class="c3"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <div
      class="c4"
    >
      <input
        class="c5"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="c6"
  />
</div>
`;
